src/sql/magic_cmd.py

```diff
@@ -90,218 +80,16 @@ def execute(self, cmd_name="", others="", cell="", local_ns=None):
-
-            return report
-
-        elif cmd_name == "explore":
-            parser = CmdParser()
-            parser.add_argument(
-                "-t", "--table", type=str, help="Table name", required=True
-            )
-            args = parser.parse_args(others)
-
-            table_widget = TableWidget(args.table)
-            display(table_widget)
-
-        elif cmd_name == "snippets":
-            return sqlcmd_snippets(others)
-
-
-def return_test_results(args, conn, query):
-    try:
-        columns = []
-        column_data = conn.execute(text(query)).cursor.description
-        res = conn.execute(text(query)).fetchall()
-        for column in column_data:
-            columns.append(column[0])
-        res = [columns, *res]
-        return res
-    except Exception as e:
-        if "column" in str(e):
-            raise exceptions.UsageError(
-                f"Referenced column '{args.column}' not found!"
-            ) from e
-
-
-def run_each_individually(args, conn):
-    base_query = select("*").from_(args.table)
-
-    storage = {}
-
-    if args.greater:
-        where = condition(args.column + "<=" + args.greater)
-        current_query = base_query.where(where).sql()
-
-        res = return_test_results(args, conn, query=current_query)
-
-        if res is not None:
-            storage["greater"] = res
-    if args.greater_or_equal:
-        where = condition(args.column + "<" + args.greater_or_equal)
-
-        current_query = base_query.where(where).sql()
-
-        res = return_test_results(args, conn, query=current_query)
-
-        if res is not None:
-            storage["greater_or_equal"] = res
-
-    if args.less_than_or_equal:
-        where = condition(args.column + ">" + args.less_than_or_equal)
-        current_query = base_query.where(where).sql()
-
-        res = return_test_results(args, conn, query=current_query)
-
-        if res is not None:
-            storage["less_than_or_equal"] = res
-    if args.less_than:
-        where = condition(args.column + ">=" + args.less_than)
-        current_query = base_query.where(where).sql()
-
-        res = return_test_results(args, conn, query=current_query)
-
-        if res is not None:
-            storage["less_than"] = res
-    if args.no_nulls:
-        where = condition("{} is NULL".format(args.column))
-        current_query = base_query.where(where).sql()
-
-        res = return_test_results(args, conn, query=current_query)
-
-        if res is not None:
-            storage["null"] = res
 
-    return storage
+        router = {
+            "tables": execute_tables_command,
+            "columns": execute_columns_command,
+            "test": execute_test_command,
+            "profile": execute_profile_command,
+            "explore": execute_expolore_command,
+            "snippets": execute_snippets_command,
+        }
+
+        cmd = router.get(cmd_name)
+        if cmd:
+            return cmd(others)
```

Comment:
what happens if I run `%sqlcmd no-a-command`? let's ensure it throws an appropriate error message and that we have a unit test to check such message

Reply:
@edublancas i checked this part. The `_validate_execute_inputs` function takes care of the validation of commands, when we run something like `%sqlcmd no-a-command` we get an error message output
<img width="1048" alt="Screenshot 2023-06-23 at 6 00 04 PM" src="https://github.com/ploomber/jupysql/assets/69951190/b67afdda-4c26-48ff-9c06-14fdfd2143e6">

The unit test for this is present in `test_magic_cmd.py` called `test_error`.

Reply:
great